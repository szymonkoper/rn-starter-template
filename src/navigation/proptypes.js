import PropTypes from 'prop-types'

export default PropTypes.shape({
  getParam: PropTypes.func.isRequired,
  navigate: PropTypes.func.isRequired,
  pop: PropTypes.func.isRequired,
  popToTop: PropTypes.func.isRequired,
  replace: PropTypes.func.isRequired
})
