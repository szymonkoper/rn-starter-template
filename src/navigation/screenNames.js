export const Comics = 'Comics'
export const Settings = 'Settings'
