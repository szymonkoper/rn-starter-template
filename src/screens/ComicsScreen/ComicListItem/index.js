export { default, ComicListItemSeparator } from './ComicListItem'
