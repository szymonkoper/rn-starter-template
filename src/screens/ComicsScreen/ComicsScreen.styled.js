import styled from 'styled-components/native'
import LottieView from 'lottie-react-native'
import { Animated } from 'react-native'
import { backgroundColorSecondary, negativeColor, white } from 'style/colors'
import { fontSizeS, itemPadding } from 'style/dimensions'
import Text from 'style/components/Text'

export const ScreenContainer = styled.SafeAreaView`
  background-color: ${backgroundColorSecondary};
  flex: 1;
`

export const ComicsFlatList = styled.FlatList.attrs(() => ({
  contentContainerStyle: {
    paddingVertical: itemPadding
  }
}))`
  flex: 1;
`

export const EmptyStateIndicator = styled(LottieView).attrs(() => ({
  autoPlay: true,
  loop: true,
  source: require('lottie/empty.json')
}))`
  width: 200px;
  height: 200px;
`

export const CenterFullAbsolute = styled.View.attrs(() => ({
  pointerEvents: 'none'
}))`
  align-items: center;
  height: 100%;
  justify-content: center;
  position: absolute;
  width: 100%;
`

export const EmptyStateText = styled(Text)`
  font-size: ${fontSizeS};
`

export const FloatingErrorContainer = styled(Animated.View)`
  align-items: center;
  background-color: ${negativeColor};
  border-radius: ${itemPadding}px;
  bottom: 0px;
  elevation: 1;
  height: 40px;
  justify-content: center;
  position: absolute;
  width: 100%;
`

export const FloatingErrorText = styled(Text).attrs(() => ({
  numberOfLines: 1,
  ellipsizeMode: 'tail'
}))`
  color: ${white};
  font-size: ${fontSizeS};
`
