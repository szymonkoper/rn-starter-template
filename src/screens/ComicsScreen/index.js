export { default } from './ComicsScreen.container'
