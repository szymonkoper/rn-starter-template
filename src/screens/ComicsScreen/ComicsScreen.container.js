import { connect } from 'react-redux'
import ComicsScreen from './ComicsScreen'

export default connect(null, null)(ComicsScreen)
