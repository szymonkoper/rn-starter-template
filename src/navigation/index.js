import { createAppContainer } from 'react-navigation'
import createNativeStackNavigator from 'react-native-screens/createNativeStackNavigator'

import screens, { initialRouteName } from './screens'

const options = {
  initialRouteName
}

const Navigator = createNativeStackNavigator(screens, options)
const NavigationContainer = createAppContainer(Navigator)

export { initialRouteName }

export default NavigationContainer
