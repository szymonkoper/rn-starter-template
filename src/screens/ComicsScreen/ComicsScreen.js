import React, { useEffect, useState } from 'react'
import { Animated } from 'react-native'
import PropTypes from 'prop-types'
import { useTranslation } from 'react-i18next'
import * as ScreenNames from 'navigation/screenNames'
import navigationShape from 'navigation/proptypes'
import { comicsShape } from 'reducers/api/xkcdRest/proptypes'
import HeaderButton from 'style/components/HeaderButton'
import LoadingIndicator from 'style/components/LoadingIndicator'
import {
  CenterFullAbsolute,
  ComicsFlatList,
  EmptyStateIndicator,
  EmptyStateText,
  FloatingErrorContainer,
  FloatingErrorText,
  ScreenContainer
} from './ComicsScreen.styled'
import ComicListItem, { ComicListItemSeparator } from './ComicListItem'

const TOAST_DURATION = 3000

const ANIMATION_SHOW = {
  useNativeDriver: true,
  duration: 300,
  toValue: 1
}

const ANIMATION_HIDE = { ...ANIMATION_SHOW, toValue: 0 }

const ComicsScreen = ({
  comics,
  hasNext,
  error,
  getNextComics,
  loading,
  navigation,
  resetComics
}) => {
  const { t } = useTranslation()

  useEffect(() => {
    getNextComics()
  }, [])

  const [toastShowAnimation] = useState(new Animated.Value(0))
  const [toastHideTimeout, setToastHideTimeout] = useState(null)

  useEffect(() => {
    if (error) {
      clearTimeout(toastHideTimeout)

      Animated.timing(toastShowAnimation, ANIMATION_SHOW).start()

      setToastHideTimeout(
        setTimeout(() => {
          Animated.timing(toastShowAnimation, ANIMATION_HIDE).start()
        }, TOAST_DURATION)
      )
    }
  }, [error])

  const onComicPress = pressedNum => {
    const { img } = comics.find(({ num }) => num === pressedNum)

    navigation.navigate(ScreenNames.ComicDetail, { imgSource: img })
  }

  return (
    <ScreenContainer>
      <ComicsFlatList
        data={comics}
        renderItem={({ item: { img, num, title } }) => (
          <ComicListItem
            imgSource={img}
            onPress={() => onComicPress(num)}
            title={title}
          />
        )}
        keyExtractor={({ num }) => num.toString()}
        onEndReached={() => {
          if (hasNext) getNextComics()
        }}
        onEndReachedThreshold={0.5}
        onRefresh={async () => {
          await resetComics()
          getNextComics()
        }}
        refreshing={false}
        ItemSeparatorComponent={ComicListItemSeparator}
      />

      {comics.length === 0 && (
        <CenterFullAbsolute>
          {loading ? (
            <LoadingIndicator />
          ) : (
            <>
              <EmptyStateIndicator />
              <EmptyStateText>
                {error ? t('Error.NoNetwork') : t('Comics.EmptyStateMessage')}
              </EmptyStateText>
            </>
          )}
        </CenterFullAbsolute>
      )}

      <FloatingErrorContainer
        style={{
          transform: [
            {
              translateY: toastShowAnimation.interpolate({
                inputRange: [0, 1],
                outputRange: [40, 0]
              })
            },
            {
              scale: toastShowAnimation.interpolate({
                inputRange: [0.4, 0.9, 1],
                outputRange: [1, 1.5, 1],
                extrapolate: 'clamp'
              })
            }
          ]
        }}
      >
        <FloatingErrorText>{t('Error.NoNetwork')}</FloatingErrorText>
      </FloatingErrorContainer>
    </ScreenContainer>
  )
}

ComicsScreen.propTypes = {
  error: PropTypes.shape({ code: PropTypes.number }),
  comics: comicsShape.isRequired,
  loading: PropTypes.bool.isRequired,
  hasNext: PropTypes.bool.isRequired,
  getNextComics: PropTypes.func.isRequired,
  resetComics: PropTypes.func.isRequired,
  navigation: navigationShape.isRequired
}

ComicsScreen.defaultProps = {
  error: null
}

ComicsScreen.navigationOptions = ({ navigation, screenProps: { t } }) => ({
  title: t('Comics.HeaderTitle'),
  headerRight: () => (
    <HeaderButton
      onPress={() => navigation.navigate(ScreenNames.Settings)}
      title={t('Settings.HeaderTitle')}
    />
  )
})

export default ComicsScreen
