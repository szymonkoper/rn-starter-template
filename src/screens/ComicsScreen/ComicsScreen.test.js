import React from 'react'
import { ScrollView } from 'react-native'
import { fireEvent, render } from 'react-native-testing-library'
import { I18nextProvider } from 'react-i18next'
import i18next from 'i18n/__test__/testInitialize'
import * as ScreenNames from 'navigation/screenNames'
import ComicsScreen from './ComicsScreen'

describe('ComicsScreen', () => {
  let props = null
  let renderedScreen = null

  beforeEach(() => {
    props = {
      comics: [
        { num: 123, title: 'Comic title', img: 'http://example.com/image.jpg' }
      ],
      hasNext: true,
      error: null,
      getNextComics: jest.fn(),
      loading: false,
      resetComics: jest.fn(),
      navigation: {
        getParam: jest.fn(),
        navigate: jest.fn(),
        pop: jest.fn(),
        popToTop: jest.fn(),
        replace: jest.fn()
      }
    }

    renderedScreen = render(
      <I18nextProvider i18n={i18next}>
        {/* eslint-disable-next-line react/jsx-props-no-spreading */}
        <ComicsScreen {...props} />
      </I18nextProvider>
    )
  })

  it('renders correctly', () => {
    const { toJSON } = renderedScreen
    expect(toJSON()).toMatchSnapshot()
  })

  it('fetches data on start', () => {
    expect(props.getNextComics).toHaveBeenCalled()
  })

  it('navigates to ComicDetail on comic item press', () => {
    const { getByText } = renderedScreen

    const { title, img } = props.comics[0]

    fireEvent.press(getByText(title))

    expect(props.navigation.navigate).toHaveBeenCalledWith(
      ScreenNames.ComicDetail,
      {
        imgSource: img
      }
    )
  })
})
