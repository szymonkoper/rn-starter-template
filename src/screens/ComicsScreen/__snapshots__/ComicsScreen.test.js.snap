// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComicsScreen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#e0e0e0",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <RCTScrollView
    ItemSeparatorComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "displayName": "Styled(View)",
        "inlineStyle": InlineStyle {
          "rules": Array [
            "
  height: ",
            "16",
            "px;
",
          ],
        },
        "render": [Function],
        "styledComponentId": "StyledNativeComponent",
        "target": [Function],
        "withComponent": [Function],
      }
    }
    contentContainerStyle={
      Object {
        "paddingVertical": 16,
      }
    }
    data={
      Array [
        Object {
          "img": "http://example.com/image.jpg",
          "num": 123,
          "title": "Comic title",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.5}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 16,
              "flexDirection": "row",
              "height": 100,
              "marginHorizontal": 16,
              "opacity": 1,
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "column",
                  "width": "30%",
                },
              ]
            }
          >
            <Image
              onLoadEnd={[Function]}
              onLoadStart={[Function]}
              resizeMode="contain"
              source={
                Object {
                  "uri": "http://example.com/image.jpg",
                }
              }
              style={
                Array [
                  Object {
                    "height": "100%",
                    "width": "100%",
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "aspectRatio": 1,
                  },
                  undefined,
                  Array [
                    Object {
                      "height": "100%",
                      "position": "absolute",
                      "width": "100%",
                    },
                  ],
                ]
              }
            >
              <View
                autoPlay={true}
                enableMergePathsAndroidForKitKatAndAbove={false}
                loop={true}
                onAnimationFinish={[Function]}
                progress={0}
                resizeMode="contain"
                sourceJson="{\\"v\\":\\"4.6.0\\",\\"fr\\":29.9700012207031,\\"ip\\":0,\\"op\\":49.0000019958109,\\"w\\":200,\\"h\\":200,\\"nm\\":\\"loading_ring_medium\\",\\"ddd\\":0,\\"assets\\":[],\\"layers\\":[{\\"ddd\\":0,\\"ind\\":1,\\"ty\\":4,\\"nm\\":\\"green ring 1\\",\\"ks\\":{\\"o\\":{\\"a\\":0,\\"k\\":100},\\"r\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.833],\\"y\\":[0.833]},\\"o\\":{\\"x\\":[0.167],\\"y\\":[0.167]},\\"n\\":[\\"0p833_0p833_0p167_0p167\\"],\\"t\\":0,\\"s\\":[0],\\"e\\":[360]},{\\"t\\":49.0000019958109}]},\\"p\\":{\\"a\\":0,\\"k\\":[100,100,0]},\\"a\\":{\\"a\\":0,\\"k\\":[0,0,0]},\\"s\\":{\\"a\\":0,\\"k\\":[200,200,100]}},\\"ao\\":0,\\"shapes\\":[{\\"ty\\":\\"gr\\",\\"it\\":[{\\"d\\":1,\\"ty\\":\\"el\\",\\"s\\":{\\"a\\":0,\\"k\\":[54,54]},\\"p\\":{\\"a\\":0,\\"k\\":[0,0]},\\"nm\\":\\"Ellipse Path 1\\",\\"mn\\":\\"ADBE Vector Shape - Ellipse\\"},{\\"ty\\":\\"st\\",\\"c\\":{\\"a\\":0,\\"k\\":[0.2235294117647059,0.35294117647058826,1,1]},\\"o\\":{\\"a\\":0,\\"k\\":100},\\"w\\":{\\"a\\":0,\\"k\\":6},\\"lc\\":2,\\"lj\\":1,\\"ml\\":4,\\"nm\\":\\"Stroke 1\\",\\"mn\\":\\"ADBE Vector Graphic - Stroke\\"},{\\"ty\\":\\"tr\\",\\"p\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":2},\\"a\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":1},\\"s\\":{\\"a\\":0,\\"k\\":[100,100],\\"ix\\":3},\\"r\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":6},\\"o\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":7},\\"sk\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":4},\\"sa\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":5},\\"nm\\":\\"Transform\\"}],\\"nm\\":\\"Ellipse 1\\",\\"np\\":3,\\"mn\\":\\"ADBE Vector Group\\"},{\\"ty\\":\\"tm\\",\\"s\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.667],\\"y\\":[1]},\\"o\\":{\\"x\\":[0.644],\\"y\\":[0]},\\"n\\":[\\"0p667_1_0p644_0\\"],\\"t\\":10,\\"s\\":[0],\\"e\\":[100]},{\\"t\\":50.0000020365418}],\\"ix\\":1},\\"e\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.667],\\"y\\":[1]},\\"o\\":{\\"x\\":[0.333],\\"y\\":[0]},\\"n\\":[\\"0p667_1_0p333_0\\"],\\"t\\":-1,\\"s\\":[0],\\"e\\":[100]},{\\"t\\":37.0000015070409}],\\"ix\\":2},\\"o\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":3},\\"m\\":1,\\"ix\\":2,\\"nm\\":\\"Trim Paths 1\\",\\"mn\\":\\"ADBE Vector Filter - Trim\\"}],\\"ip\\":0,\\"op\\":50.0000020365418,\\"st\\":-1.00000004073083,\\"bm\\":0,\\"sr\\":1},{\\"ddd\\":0,\\"ind\\":2,\\"ty\\":4,\\"nm\\":\\"flamingo ring 3\\",\\"parent\\":1,\\"ks\\":{\\"o\\":{\\"a\\":0,\\"k\\":100},\\"r\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.785],\\"y\\":[1]},\\"o\\":{\\"x\\":[1],\\"y\\":[0]},\\"n\\":[\\"0p785_1_1_0\\"],\\"t\\":17,\\"s\\":[14.2],\\"e\\":[360]},{\\"t\\":50.0000020365418}]},\\"p\\":{\\"a\\":0,\\"k\\":[0,0,0]},\\"a\\":{\\"a\\":0,\\"k\\":[0,0,0]},\\"s\\":{\\"a\\":0,\\"k\\":[100,100,100]}},\\"ao\\":0,\\"shapes\\":[{\\"ty\\":\\"gr\\",\\"it\\":[{\\"d\\":1,\\"ty\\":\\"el\\",\\"s\\":{\\"a\\":0,\\"k\\":[54,54]},\\"p\\":{\\"a\\":0,\\"k\\":[0,0]},\\"nm\\":\\"Ellipse Path 1\\",\\"mn\\":\\"ADBE Vector Shape - Ellipse\\"},{\\"ty\\":\\"st\\",\\"c\\":{\\"a\\":0,\\"k\\":[0.08627450980392157,0.8196078431372549,0.9019607843137255,1]},\\"o\\":{\\"a\\":0,\\"k\\":100},\\"w\\":{\\"a\\":0,\\"k\\":6},\\"lc\\":2,\\"lj\\":1,\\"ml\\":4,\\"nm\\":\\"Stroke 1\\",\\"mn\\":\\"ADBE Vector Graphic - Stroke\\"},{\\"ty\\":\\"tr\\",\\"p\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":2},\\"a\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":1},\\"s\\":{\\"a\\":0,\\"k\\":[100,100],\\"ix\\":3},\\"r\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":6},\\"o\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":7},\\"sk\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":4},\\"sa\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":5},\\"nm\\":\\"Transform\\"}],\\"nm\\":\\"Ellipse 1\\",\\"np\\":3,\\"mn\\":\\"ADBE Vector Group\\"},{\\"ty\\":\\"tm\\",\\"s\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.833],\\"y\\":[0.833]},\\"o\\":{\\"x\\":[0.167],\\"y\\":[0.167]},\\"n\\":[\\"0p833_0p833_0p167_0p167\\"],\\"t\\":42,\\"s\\":[0],\\"e\\":[1]},{\\"t\\":44.0000017921567}],\\"ix\\":1},\\"e\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.833],\\"y\\":[1]},\\"o\\":{\\"x\\":[0.333],\\"y\\":[0]},\\"n\\":[\\"0p833_1_0p333_0\\"],\\"t\\":20,\\"s\\":[0],\\"e\\":[1]},{\\"t\\":44.0000017921567}],\\"ix\\":2},\\"o\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":3},\\"m\\":1,\\"ix\\":2,\\"nm\\":\\"Trim Paths 1\\",\\"mn\\":\\"ADBE Vector Filter - Trim\\"}],\\"ip\\":17.0000006924242,\\"op\\":44.0000017921567,\\"st\\":-1.00000004073083,\\"bm\\":0,\\"sr\\":1},{\\"ddd\\":0,\\"ind\\":3,\\"ty\\":4,\\"nm\\":\\"flamingo ring 2\\",\\"parent\\":1,\\"ks\\":{\\"o\\":{\\"a\\":0,\\"k\\":100},\\"r\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.612],\\"y\\":[1]},\\"o\\":{\\"x\\":[1],\\"y\\":[0]},\\"n\\":[\\"0p612_1_1_0\\"],\\"t\\":17,\\"s\\":[14.2],\\"e\\":[360]},{\\"t\\":50.0000020365418}]},\\"p\\":{\\"a\\":0,\\"k\\":[0,0,0]},\\"a\\":{\\"a\\":0,\\"k\\":[0,0,0]},\\"s\\":{\\"a\\":0,\\"k\\":[100,100,100]}},\\"ao\\":0,\\"shapes\\":[{\\"ty\\":\\"gr\\",\\"it\\":[{\\"d\\":1,\\"ty\\":\\"el\\",\\"s\\":{\\"a\\":0,\\"k\\":[54,54]},\\"p\\":{\\"a\\":0,\\"k\\":[0,0]},\\"nm\\":\\"Ellipse Path 1\\",\\"mn\\":\\"ADBE Vector Shape - Ellipse\\"},{\\"ty\\":\\"st\\",\\"c\\":{\\"a\\":0,\\"k\\":[0.08627450980392157,0.8196078431372549,0.9019607843137255,1]},\\"o\\":{\\"a\\":0,\\"k\\":100},\\"w\\":{\\"a\\":0,\\"k\\":6},\\"lc\\":2,\\"lj\\":1,\\"ml\\":4,\\"nm\\":\\"Stroke 1\\",\\"mn\\":\\"ADBE Vector Graphic - Stroke\\"},{\\"ty\\":\\"tr\\",\\"p\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":2},\\"a\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":1},\\"s\\":{\\"a\\":0,\\"k\\":[100,100],\\"ix\\":3},\\"r\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":6},\\"o\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":7},\\"sk\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":4},\\"sa\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":5},\\"nm\\":\\"Transform\\"}],\\"nm\\":\\"Ellipse 1\\",\\"np\\":3,\\"mn\\":\\"ADBE Vector Group\\"},{\\"ty\\":\\"tm\\",\\"s\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.833],\\"y\\":[0.833]},\\"o\\":{\\"x\\":[0.167],\\"y\\":[0.167]},\\"n\\":[\\"0p833_0p833_0p167_0p167\\"],\\"t\\":42,\\"s\\":[0],\\"e\\":[13.7]},{\\"t\\":44.0000017921567}],\\"ix\\":1},\\"e\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.833],\\"y\\":[1]},\\"o\\":{\\"x\\":[0.333],\\"y\\":[0]},\\"n\\":[\\"0p833_1_0p333_0\\"],\\"t\\":20,\\"s\\":[0],\\"e\\":[13.7]},{\\"t\\":44.0000017921567}],\\"ix\\":2},\\"o\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":3},\\"m\\":1,\\"ix\\":2,\\"nm\\":\\"Trim Paths 1\\",\\"mn\\":\\"ADBE Vector Filter - Trim\\"}],\\"ip\\":17.0000006924242,\\"op\\":44.0000017921567,\\"st\\":-1.00000004073083,\\"bm\\":0,\\"sr\\":1},{\\"ddd\\":0,\\"ind\\":4,\\"ty\\":4,\\"nm\\":\\"flaming ring 1\\",\\"parent\\":1,\\"ks\\":{\\"o\\":{\\"a\\":0,\\"k\\":100},\\"r\\":{\\"a\\":0,\\"k\\":0},\\"p\\":{\\"a\\":0,\\"k\\":[0,0,0]},\\"a\\":{\\"a\\":0,\\"k\\":[0,0,0]},\\"s\\":{\\"a\\":0,\\"k\\":[100,100,100]}},\\"ao\\":0,\\"shapes\\":[{\\"ty\\":\\"gr\\",\\"it\\":[{\\"d\\":1,\\"ty\\":\\"el\\",\\"s\\":{\\"a\\":0,\\"k\\":[54,54]},\\"p\\":{\\"a\\":0,\\"k\\":[0,0]},\\"nm\\":\\"Ellipse Path 1\\",\\"mn\\":\\"ADBE Vector Shape - Ellipse\\"},{\\"ty\\":\\"st\\",\\"c\\":{\\"a\\":0,\\"k\\":[0.08627450980392157,0.8196078431372549,0.9019607843137255,1]},\\"o\\":{\\"a\\":0,\\"k\\":100},\\"w\\":{\\"a\\":0,\\"k\\":6},\\"lc\\":2,\\"lj\\":1,\\"ml\\":4,\\"nm\\":\\"Stroke 1\\",\\"mn\\":\\"ADBE Vector Graphic - Stroke\\"},{\\"ty\\":\\"tr\\",\\"p\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":2},\\"a\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":1},\\"s\\":{\\"a\\":0,\\"k\\":[100,100],\\"ix\\":3},\\"r\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":6},\\"o\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":7},\\"sk\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":4},\\"sa\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":5},\\"nm\\":\\"Transform\\"}],\\"nm\\":\\"Ellipse 1\\",\\"np\\":3,\\"mn\\":\\"ADBE Vector Group\\"},{\\"ty\\":\\"tm\\",\\"s\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.667],\\"y\\":[1]},\\"o\\":{\\"x\\":[1],\\"y\\":[0]},\\"n\\":[\\"0p667_1_1_0\\"],\\"t\\":8,\\"s\\":[0],\\"e\\":[100]},{\\"t\\":48.0000019550801}],\\"ix\\":1},\\"e\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.667],\\"y\\":[1]},\\"o\\":{\\"x\\":[1],\\"y\\":[0]},\\"n\\":[\\"0p667_1_1_0\\"],\\"t\\":-1,\\"s\\":[0],\\"e\\":[100]},{\\"t\\":37.0000015070409}],\\"ix\\":2},\\"o\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":3},\\"m\\":1,\\"ix\\":2,\\"nm\\":\\"Trim Paths 1\\",\\"mn\\":\\"ADBE Vector Filter - Trim\\"}],\\"ip\\":44.0000017921567,\\"op\\":44.0000017921567,\\"st\\":-1.00000004073083,\\"bm\\":0,\\"sr\\":1,\\"hidden\\":15.0000006109625}]}"
                speed={1}
                style={
                  Object {
                    "aspectRatio": 1,
                    "height": "100%",
                    "position": "absolute",
                    "width": "100%",
                  }
                }
              />
            </View>
          </View>
          <Text
            ellipsizeMode="tail"
            numberOfLines={2}
            style={
              Array [
                Object {
                  "color": "#000",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontSize": 26,
                  "marginLeft": 16,
                  "paddingBottom": 4,
                  "paddingLeft": 4,
                  "paddingRight": 4,
                  "paddingTop": 4,
                  "textAlign": "center",
                },
              ]
            }
          >
            Comic title
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#a30b37",
        "borderRadius": 16,
        "bottom": 0,
        "elevation": 1,
        "height": 40,
        "justifyContent": "center",
        "position": "absolute",
        "transform": Array [
          Object {
            "translateY": 40,
          },
          Object {
            "scale": 1,
          },
        ],
        "width": "100%",
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 18,
          },
        ]
      }
    >
      Error.NoNetwork
    </Text>
  </View>
</RCTSafeAreaView>
`;
